desk/app/channels.hoon

```diff
@@ -1343,17 +1343,24 @@
     ^+  ca-core
     =.  ca-core
       %^  give  %fact  ~
-      ::  give result if we authored the content,
-      ::  or if it's readably by the requester.
+      ::  give result if it's readable by the requester,
+      ::  or if we pinned it intentionally
       ::
-      ?.  ?|  ?~  post=(get:on-v-posts:c posts.channel p.plan)  |
-              ?~  u.post  |
-              =(our.bowl author.u.u.post)
-            ::
-              (can-read:ca-perms src.bowl)
-          ==
-        channel-denied+!>(~)
-      (said-2:utils nest plan posts.channel)
+      =;  share=?
+        ?.  share
+          channel-denied+!>(~)
+        (said-2:utils nest plan posts.channel)
+      ?:  (can-read:ca-perms src.bowl)  &
+      ?^  q.plan  |  ::NOTE  expose/+grab-post doesn't support replies
+      ::  we need to grab the post first before we can check whether it's
+      ::  pinned, because its kind appears in the reference path...
```

Comment:
This suggests that:

1. The reference path format is garbage (but we already knew that).
2. Making expose agent track exposed/pinned messages by `$cite` reference was maybe a mistake. Given that we only do this for messages right now (not other kinds of references, like groups or apps) it should maybe just be storing `$plan`s or similar in state, simpler message pointers that are easier to construct/work with.

But, both of those are out of scope for here, I'd say.